paddle/phi/kernels/impl/set_value_kernel_impl.h

```diff
@@ -185,16 +184,32 @@ void SetValueImpl(const Context& dev_ctx,
   // is [3], which is right.
 
   slice_tensor.Resize(slice_dims_for_assign);
+
   CheckIsDimsMatch(slice_dims_for_assign, value.dims());
-  // ElementwiseComputeEx can do broadcasting
-  funcs::ElementwiseCompute<funcs::SubtractFunctor<T>, T>(
-      dev_ctx,
-      slice_tensor,
-      value,
-      -1,
-      funcs::SubtractFunctor<T>(),
-      &slice_tensor);
 
+  bool is_gpu_place = dev_ctx.GetPlace().GetType() == phi::AllocationType::GPU;
+  if (is_gpu_place || slice_tensor.dims().size() >= value.dims().size()) {
+    // ElementwiseComputeEx can do broadcasting in two cases:
+    // 1. The place is GPU.
+    // 2. The place is CPU, and the 'x' does not need broadcast.
+    // Please see the note in
+    // paddle/fluid/operators/elementwise/elementwise_op_function.h
```

Comment:
I understand that this is only a workaround solution, so need to add TODO to explain the futrue work.